### PR TITLE
Cypress testing

### PR DESCRIPTION
**Pull request by oorestisime**
#### What does this PR do?

TLDR: Adds support for e2e testing and runs visual regression tests on storybook

* configures cypress: this should enable running other kind of tests of integration tests as well as cross-browser ones
* add cypress-image-snapshot with a few stories (the idea is that it visits the storybook stories, snapshots them and verifies visual regression)
* migrates everything to circle ci since it allows to download artifacts (this case the diffs between expected and produces snapshots). Pricing for circleci is free for 1 concurrent job and 1000 runtime minutes per month. that equals to 200-250 builds per month which is like 5-8 builds per day. Then it gets to 50$ per month for 2 concurrent jobs and unlimited minutes per month. I don't know if they offer any packages for open source projects. Travis can not yet be ditched since i haven't added the release task (which i am not sure i can :D )

#### Notes
* this is basically another round of feedback.
* haven't added cross-browser tests yet but it shouldn't be too hard either.
* this looks great as well https://storybook.js.org/testing/structural-testing/ and maybe it should be considered. there are a lot of stories that would make good unit-tests
* still a lot of component and stories missing. i am still troubling to find a reliable way to avoid to have to write a list of stories by hand. 
* most of commits are tests, playing (with the threshold) and going back and forth. if there is a lot of noise i can rebase and fixup everything

#### What are the relevant issues?
#2571

### CODE REVIEW

**Comment by ShimiSun** [comment]
@oorestisime really nice work here! can you list all the action items related that are missing from this PR to get down with the complete desired functionality? 
What will be the ramifications for merging this PR on future maintenance processes?

**Comment by oorestisime** [comment]
Needing work:

- [ ] it seems the release process is on travis? if so it needs to be ported on circleci, unless we keep both of them in place which can be an option until all migration happens

Most important things to consider here:

- [ ] plan for circleci, if going on paid subscription, if anyone can find out whether they have free plans for open source

- [ ] if this should run on each PR, keeping in mind that it takes 2 minutes for the e2e tests but there are not a lot at the moment. i can see the minutes getting up to 5 minutes easily. IMHO this should not be a problem, the are rare occasions in need to merge a PR in 15 minutes or so, and i guess there's always the possibility to merge without the go on from the CI

Less important things:

- [ ] whether it is in the scope of this PR to add more regression tests

- [ ] whether it is interesting to have https://storybook.js.org/testing/structural-testing/
